contrib/kafka/cmd/dispatcher/main.go

```diff
@@ -68,17 +55,10 @@ func main() {
 		logger.Fatal("Unable to add kafkaDispatcher", zap.Error(err))
 	}
 
-	kc, err := kubernetes.NewForConfig(mgr.GetConfig())
-	if err != nil {
-		logger.Fatal("unable to create kubernetes client.", zap.Error(err))
-	}
-
-	cmw, err := watcher.NewWatcher(logger, kc, configMapNamespace, configMapName, kafkaDispatcher.UpdateConfig)
+	v1alpha1.AddToScheme(mgr.GetScheme())
+	channelwatcher.New(mgr, logger, channelwatcher.UpdateConfigWatchHandler(kafkaDispatcher.UpdateConfig, shouldWatch))
 	if err != nil {
```

Comment:
The previous line doesn't create an error, should it? If not, then this is unreachable code (err was last set on line 40 and checked on 41).

Reply:
both lines 58 and 59 create errors. I missed them. Will fix.